### PR TITLE
Make the namespace of doctests per-module

### PR DESCRIPTION
**Pull request by lucafavatellawork**
This plants the solution discussed at https://github.com/opensparkl/sse_cli/pull/77#discussion_r425625188 re how make the doctest namespace non-globally scoped without relying on file names.

Please refer to the sequence of commits - with their messages and CI outcome - for details.

The CI fails. It is expected as this CI moves the test files. I reasonably proved that this PR will pass CI once merged by temporarily using the local files in the commit history then reverting that change in a recent commit.

### CODE REVIEW

File: docs/basic/tutorial_rest_python.rst [hunk 1]
```diff
@@ -13,14 +13,14 @@ Let's see how we can use a Python module as the logic behind an SSE configuratio
...
 
 There are a couple of differences:
 
 - We replaced the ``Expr`` service with a service called ``Rest``. This service uses the ``rest`` extension
 - The request operations formerly backed by ``Expr`` are now on the ``Rest`` service
 - The notify and solicit operations also specify this new service as their client
-- The markup contains no Erlang expressions. The logic is in a separate `Python module <https://github.com/opensparkl/sse_cli/blob/master/docs/examples/hello-world-rest/rest_impl.py>`_
+- The markup contains no Erlang expressions. The logic is in a separate `Python module <https://github.com/opensparkl/sse_cli/blob/master/docs/basic/examples/hello-world-rest/rest_impl.py>`_
 
 Let's put our sample code in place.
```
Comment (miklosduma) [inline docs/basic/tutorial_rest_python.rst:25]: This file also has a comment:

 ``EXAMPLES_FOLDER`` is the ``docs/examples`` folder of this repository.

This I think is the only reference missed.

Comment (lucafavatellawork) [reply docs/basic/tutorial_rest_python.rst:25]: Addressed in new commit.

**Comment by lucafavatellawork** [comment]
> Might need to merge first using local reference to the examples though

You mean https://github.com/opensparkl/sse_cli/pull/79 ?

**Comment by lucafavatellawork** [comment]
@miklosduma Once this is merged I will rebase #80

**Pull request merged**